### PR TITLE
Support negative axis indexing for `axis` param in reduction ops

### LINKED ISSUE #477: Feature Request: Support python-style negative axis indexing.
For example, in tensorflow, `tf.reduce_mean(x, axis=-1)` means average over the last dimension. deeplearn.js currently doesn't support negative axis values. It would be nice to have that support.

Functions that may also benefit from this indexing are: sum, max, min.

### PR DESCRIPTION
**Pull request by dsmilkov**
Fixes #477. See #477 for detail.

---
This change is [Reviewable](https://reviewable.io/reviews/pair-code/deeplearnjs/485)

### CODE REVIEW

**Comment by nsthorat** [comment]
:lgtm_strong:

---

Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pair-code/deeplearnjs/485#-:-L1TkkRq54ZmtZMTVkSC:b-jbsf8z)*
